src/site/lab5.rst

```diff
@@ -0,0 +1,245 @@
+******
+Lab #5
+******
+
+* Refer to the topic pages for this lab
+    * You will find many of the answers to your questions there
+* Feel free to use your laptop if you have it
+* I strongly encourage you to work with others in the lab
+* Ensure I have recorded your completion. Failure to do so will result in a grade of 0
+* When you get stuck, do me a favour and ask those sitting around you for help
+    * I want people to get used to working together in the labs
+    * Peer teaching and peer learning is super effective
+
+
+Making a Linked Structure
+==========================
+
+Start by building a linked structure. The idea is to simply have a series of ``Node`` objects.
+
+
+Node Class
+----------
+
+1. Create a project and make a class where you will put your main method
+    * Perhaps call it ``Lab5``
+
+2. Create a generic ``Node`` class
+    * Feel free to :download:`download this one <../main/java/Node.java>`
+        * If you do though, be sure to understand it and don't just copy/paste
+
+3. Do a simple test to see if it works by making an instance in your main method and using the node's methods
+
+    .. code-block:: java
+        :linenos:
+
+        Node<Integer> myNode = new Node<>(5);
+        // Use the node class' methods here
+
+
+Create the Linked Structure
+---------------------------
+
+1. Write a static method in the same class as your ``main`` called ``makeLinkedStructure`` that will
+    a. Create a linked structure containing the numbers 0 -- 9
+        * One number referenced in each node
+    b. Return a reference to the head of the linked structure
+
+    .. code-block:: java
+        :linenos:
+
+        public static Node<T> makeLinkedStructure() {
+            // Stuff
+        }
+
+
+2. Verify it works by adding this to your ``main`` method
+
+    .. code-block:: java
+        :linenos:
+
+        Node<Integer> head = makeLinkedStructure();
+        Node<Integer> currentNode = head;
+
+        while (currentNode != null) {
+            System.out.println(currentNode.getData());
+            currentNode = currentNode.getNext();
+        }
+
+
+Manipulating the Linked Structure
+=================================
+
+Start adding and removing things to the linked structure.
+
+
+Adding to the Front of the Structure
+------------------------------------
+
+1. Write a static method in the same class as your ``main`` called ``addToFront`` that will
+    a. Take a reference to the head of a linked structure and an integer to add
+    b. Insert a new node containing reference to a new integer passed to the method as a parameter
+    c. Return a reference to the new head of the linked structure
+
+    .. code-block:: java
+        :linenos:
+
+        public static Node<T> addToFront(Node<T> head, Integer toAdd) {
```

Comment:
`Integer` -> `T`. occurs in instructions and other methods as well.

Reply:
Good catch my man.